src/graphql/utils/loaderFunction.ts

```diff
@@ -0,0 +1,26 @@
+import fetchWithAbort from './fetchWithAbort'
+
+// A function returns a Promise that resolves to the values corresponding the keys
+function loaderFunction(keys: readonly string[], token = ''): Promise<Array<Promise<Object>>> {
```

Comment:
Also here it would make sense to use a generic type in order to allow the consumer to determine the return type.